ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -922,18 +921,18 @@ IndexValue markAsPermanent(StoreKey id, FileSpan fileSpan, long operationTimeMs)
    * @return the {@link IndexValue} of the ttl update record
    * @throws StoreException if there is any problem writing the index record
    */
-  private IndexValue markAsPermanent(StoreKey id, FileSpan fileSpan, MessageInfo info, long operationTimeMs,
-      short lifeVersion) throws StoreException {
+  IndexValue markAsPermanent(StoreKey id, FileSpan fileSpan, MessageInfo info, long operationTimeMs, short lifeVersion)
+      throws StoreException {
     validateFileSpan(fileSpan, true);
     boolean hasLifeVersion = IndexValue.hasLifeVersion(lifeVersion);
     IndexValue value = findKey(id);
     if (value == null && info == null) {
       throw new StoreException("Id " + id + " not present in index " + dataDir, StoreErrorCodes.ID_Not_Found);
     }
     short retrievedLifeVersion = value == null ? info.getLifeVersion() : value.getLifeVersion();
-    if (value != null && value.isFlagSet(IndexValue.Flags.Delete_Index)) {
+    if (value != null && value.isDelete()) {
       throw new StoreException("Id " + id + " deleted in index " + dataDir, StoreErrorCodes.ID_Deleted);
-    } else if (value != null && value.isFlagSet(IndexValue.Flags.Ttl_Update_Index)) {
+    } else if (value != null && value.isTTLUpdate()) {
       throw new StoreException("TTL of " + id + " already updated in index" + dataDir, StoreErrorCodes.Already_Updated);
     } else if (hasLifeVersion && retrievedLifeVersion > lifeVersion) {
```

Comment:
In this PR, `hasLifeVersion` is always false, right? (because BlobStore.handleTtlUpate passes `LIFE_VERSION_FROM_FRONTEND` into this method)

Reply:
that's right, TTL_UPDATE record will always follow a PUT or an UNDELETE.